File: sicp_1_12.swift
#!/usr/bin/env swift
// sicp exercise 1.12

func pascalElem(row: Int, col: Int) -> Int {
  return col == 1 ? 1 : col == row ? 1 : (pascalElem(row-1,col-1) + pascalElem(row-1,col))
}

func createPacalsTriangle(rows: Int) {
  for r in 1...rows {
    for c in 1...r { 
      print("\(pascalElem(r,c)) ")
    }
    println()
  }
}

createPacalsTriangle(5)

File: sicp_1_12.js
#!/usr/bin/env node

function pascalEleemnt(r,c) {
  if (c === 0) {
    return 1;
  } else if (c === r) {
    return 1;
  } else {
    return pascalEleemnt(r-1,c-1) + pascalEleemnt(r-1,c);
  }
}

console.log(pascalEleemnt(0,0));
console.log(pascalEleemnt(1,0));
console.log(pascalEleemnt(4,2));